10_datastructures_lab/01.py

```diff
@@ -1,10 +1,83 @@
-"""
-Write a program that takes two strings
-from the user and checks if they represent
-a valid user name.
-Valid users and passwords:
-    apple => red
-    lettuce => green
-    lemon => yellow
-    orange => orange
-"""
+#!/usr/bin/env python
+
+"""
+This program parse the given hosts file and return the IP address of a given computers list
+
+"""
+__author__ = 'Eabel'
+
+import sys
+import os
+import argparse
+import socket
+from collections import defaultdict
+
+
+def is_valid_ipv4_address(address):
+    try:
+        socket.inet_pton(socket.AF_INET, address)
+    except AttributeError:  # no inet_pton here, sorry
+        try:
+            socket.inet_aton(address)
+        except socket.error:
+            return False
+        return address.count('.') == 3
+    except socket.error:  # not a valid address
+        return False
+
+    return True
+
+def is_valid_ipv6_address(address):
+    try:
+        socket.inet_pton(socket.AF_INET6, address)
+    except socket.error:  # not a valid address
+        return False
+    return True
+
+def is_valid_ip_address(address):
+    return is_valid_ipv4_address or is_valid_ipv6_address(address)
+
+
+def read_hosts_file(file_path):
+    try:
+        f = open(file_path)
+        host_dict = defaultdict(str)
+        for line in f:
+            items = line.split('=')

```

Comment:
Yes but don't you want to check if the IP is valid?

Sent from my iPhone

> On 10 Jul 2016, at 22:19, ynonp <notifications@github.com> wrote:
> 
> In 10_datastructures_lab/01.py:
> 
> > +    try:
> > +        socket.inet_pton(socket.AF_INET6, address)
> > +    except socket.error:  # not a valid address
> > +        return False
> > +    return True
> > +
> > +def is_valid_ip_address(address):
> > +    return is_valid_ipv4_address or is_valid_ipv6_address(address)
> > +
> > +
> > +def read_hosts_file(file_path):
> > +    try:
> > +        f = open(file_path)
> > +        host_dict = defaultdict(str)
> > +        for line in f:
> > +            items = line.split('=')
> Since you're already inside try/except block, let's consider host, ip = split('=').
> If it fails the global except is reached and you return false.
> If it succeeds you don't need to check len(items) And you get to use the much cleaner host_dict[host] = ip.
> Regarding rstrip - IMHO it's nicer to write it as the first line in the loop, such as: line = line.rstrip('\r\n').
> 
> Connecting everything we'd get:
> 
> for line in f:
>   line = line.rstrip()
>   host, ip = line.split('=')
>   host_dict[host] = ip
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>